crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -16,6 +16,7 @@ fn push_events(world: &mut World, events: impl IntoIterator<Item = HierarchyEven
     }
 }
 
+/// Adds `child` to `parent`'s [`Children`], without checking if it is already present there.
```

Comment:
What are the implications of adding `child` without checking? Could you add a line explaining why you should really not do it?

Reply:
All I know about unchecked push is that might lead to the same entity appearing in the list twice. I don't know what problems that might cause. Mostly documented just to tell what's *not* checked, since function name `push_child_unchecked` doesn't really tell what checks it skips.

Reply:
It's probably going to result in unexpected results when removing duplicate children.

The PR needs a second approval anyway, so can we keep this unresolved to get a second opinion?

```suggestion
/// Adds `child` to `parent`'s [`Children`], without checking if it is already present there.
///
/// This might cause unexpected results when removing duplicate children.
```